crate_universe/src/splicing/splicer.rs

```diff
@@ -383,12 +368,34 @@ impl<'a> SplicerKind<'a> {
                     dot_cargo_dir.join("config.toml"),
                 ] {
                     if config.exists() {
-                        fs::remove_file(&config)?;
+                        remove_symlink(&config).with_context(|| {
+                            format!(
+                                "Failed to delete existing cargo config: {}",
+                                config.display()
+                            )
+                        })?;
                     }
                 }
             }
         }
 
+        // Make sure no other config files exist
```

Comment:
I believe cargo will traverse parents all the way up to `/` to find config files - should we maybe detect (but presumably error rather than try to delete) those too?

Reply:
Oh, I didn't know that. Yeah, the logic here should only be trying to remove unspecified config files from the Cargo workspace so they don't impact them. If there's one at the system root or something I think this should error since I otherwise don't think it'd be obvious why the generated lockfile doesn't match any other host.

Reply:
Ok. Updated and added some tests.